beacon-chain/core/blob/sidecar.go

```diff
@@ -48,8 +49,23 @@ func init() {
 		current.Mul(&current, &rootOfUnity).
 			Mod(&current, &blsModulus)
 	}
+
+	rootsOfUnity = bitReversalPermutation(rootsOfUnity)
 }
 
+// Return a copy with bit-reversed permutation. This operation is idempotent.
+// l is the array of roots of unity
+func bitReversalPermutation(l [params.FieldElementsPerBlob]bls.Fr) [params.FieldElementsPerBlob]bls.Fr {
+		var out [params.FieldElementsPerBlob]bls.Fr
+
+		for i := range l {
+			j := bits.Reverse64(uint64(i)) >> (65 - bits.Len64(params.FieldElementsPerBlob))
```

Comment:
implementors note: this makes the assumption that `params.FieldElementsPerBlob` is a power of 2 (which it is: 4096)

Reply:
Maybe we can add a panic here if that's not the case. (I can imagine a "minimal" beacon chain configuration with a tweaked `FieldElementsPerBlob`). Otherwise this may silently break.

Reply:
the array indexing will break anyway if not a power of two, but agreed, a panic is more explicit. Added